sympy/assumptions/handlers/sets.py

```diff
@@ -741,7 +741,20 @@ def _(expr, assumptions):
         if ask(Q.algebraic(expr.exp), assumptions):
             return ask(~Q.nonzero(expr.exp), assumptions)
         return
-    return expr.exp.is_Rational and ask(Q.algebraic(expr.base), assumptions)
+    if expr.base == pi:
+        if ask(Q.integer(expr.exp), assumptions) and ask(Q.positive(expr.exp), assumptions):
+            return False
+        return
+    exp_rational = ask(Q.rational(expr.exp), assumptions)
+    base_algebraic = ask(Q.algebraic(expr.base), assumptions)
+    exp_algebraic = ask(Q.algebraic(expr.exp),assumptions)
+    if base_algebraic and exp_algebraic:
+      if exp_rational:
+          return True
+      # Check based on the Gelfond-Schneider theorem:
+      # If the base is algebraic and not equal to 0 or 1, and the exponent is irrational,then the result is transcendental.
```

Comment:
Please use 4 spaces for indentation as in PEP 8.

Also there is no need for comments to have such long lines. They can be wrapped at 80 characters.